framework/src/utils/MortarUtils.C

```diff
@@ -222,11 +222,22 @@ projectQPoints3d(const Elem * const msm_elem,
         x1 += Moose::fe_lagrange_2D_shape(sub_element_type(), FIRST, n, xi) *
               primal_elem->point(sub_elem_node_indices[n]);
       auto u = x1 - x0;
+
       VectorValue<Dual2> F(u(1) * normal(2) - u(2) * normal(1),
                            u(2) * normal(0) - u(0) * normal(2),
                            u(0) * normal(1) - u(1) * normal(0));
 
-      if (MetaPhysicL::raw_value(F).norm() < 1e-12)
+      Real projection_tolerance(1e-10);
```

Comment:
So with this change and with also possibly increasing the tolerance for large displacements, we clearly have a larger tolerance than before. I am guessing based on your title post that this now allows you to run the 3D contact problems you weren't able to run before? E.g. we had too tight of a tolerance before and weren't registering "sufficiently successful" projections?

Reply:
> I am guessing based on your title post that this now allows you to run the 3D contact problems you weren't able to run before?

Yes, 3D problem with mechanical contact and curved geometry (and small displacement units resulting in large numbers)

>  E.g. we had too tight of a tolerance before and weren't registering "sufficiently successful" projections?

Exactly. As you mentioned above, tolerances will be looser in general. However, I noticed that, for a regular projection, in the middle of a curved sideset, projections weren't successful. It is true that the algorithm was achieving tighter tolerances for other cases, but my take is that this problem, since it has mechanical contact, friction, large deformation, and curved surfaces, is a better yardstick to set this tolerance. I was happy to see that this change only caused that diff for the trivial solution of LMs, so I think I'd prefer to have this change in the code to allow MOOSE to solve more "challenging" problems.